File: frontend/src/api/audit.ts
import client from "./client";

export interface AuditLog {
  id: number;
  user_id: string | null;
  username: string | null;
  action: string;
  resource_type: string | null;
  resource_id: string | null;
  ip_address: string | null;
  occurred_at: string;
}

export async function fetchAuditLogs(params?: {
  limit?: number;
  cursor?: number;
  action?: string;
  resource_type?: string;
}): Promise<AuditLog[]> {
  return (await client.get<AuditLog[]>("/audit-logs", { params })).data;
}

File: backend/app/api/schemas/audit.py
from datetime import datetime
from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: int
    user_id: str | None
    username: str | None
    action: str
    resource_type: str | None
    resource_id: str | None
    ip_address: str | None
    user_agent: str | None
    occurred_at: datetime
